src/overview/components/DateRangeSelection.jsx

```diff
@@ -1,46 +1,126 @@
-import React, { PropTypes } from 'react'
+import React, { Component, PropTypes } from 'react'
 import moment from 'moment'
+import chrono from 'chrono-node'
 import DatePicker from 'react-datepicker'
 import 'react-datepicker/dist/react-datepicker-cssmodules.css'
 
 import styles from './DateRangeSelection.css'
 
 
-const DateRangeSelection = ({
-    startDate, endDate,
-    onStartDateChange, onEndDateChange,
-}) => (
-    <div className={styles.dateRangeSelection}>
-        <DatePicker
-            className={styles.datePicker}
-            dateFormat='DD-MM-YYYY'
-            placeholderText='after..'
-            isClearable
-            selected={startDate && moment(startDate)}
-            selectsStart
-            startDate={moment(startDate || 0)}
-            endDate={moment(endDate)}
-            maxDate={moment()}
-            onChange={date => onStartDateChange(
-                date ? date.valueOf() : undefined,
-            )}
-        />
-        <DatePicker
-            className={styles.datePicker}
-            dateFormat='DD-MM-YYYY'
-            placeholderText='before..'
-            isClearable
-            selected={endDate && moment(endDate)}
-            selectsEnd
-            startDate={moment(startDate || 0)}
-            endDate={moment(endDate)}
-            maxDate={moment()}
-            onChange={date => onEndDateChange(
-                date ? date.endOf('day').valueOf() : undefined,
-            )}
-        />
-    </div>
-)
+class DateRangeSelection extends Component {
+    constructor() {
+        super()
+
+        this.state = {
+            startDateText: null,
+            endDateText: null,
+        }
+
+        this.handleKeydown = this.handleKeydown.bind(this)
+        this.submitDateChange = this.submitDateChange.bind(this)
+    }
+
+    componentDidMount() {
+        this.refs.startDateDatePicker.refs.input.refs.input.addEventListener('keydown', e => this.handleKeydown(e, 'startDate'))
+        this.refs.endDateDatePicker.refs.input.refs.input.addEventListener('keydown', e => this.handleKeydown(e, 'endDate'))
```

Comment:
I understood why you did this. If it is not possible to change the Datepicker behaviour I'd rather wrap it in a component which will do this for you (possibly using a proper way) instead of mixing concerns